test/CMakeLists.txt

```diff
@@ -67,7 +67,14 @@ macro(onedpl_add_test test_source_file switch_off_checked_iterators)
     get_filename_component(_test_name ${test_source_file} NAME)
     string(REPLACE "\.cpp" "" _test_name ${_test_name})
 
-    add_executable(${_test_name} EXCLUDE_FROM_ALL "${test_source_file}")
+    # add additional translation units to current test according to mask
```

Comment:
Do you think we will ever want the same extra translational unit (`shared_util.cpp`) to be added for multiple different tests?  

If so, it may be better to instead have some list for each multi translational unit test which CMake could parse to find out the extra translational units that are required.  With the current method, we would need multiple copies of the same file named differently to achieve that.  

Instead you could have a file `${_test_name_without_pass}_extra_TU_List.txt` which contains a list of extra source files to add.  CMake can check if that file exists and iterate through the list if it does.  

On the other hand, if we never think this is useful, it isn't worth the extra complexity.

Reply:
That is interesting. Personally I'd prefer more explicit approach contrary to "everything covered by some glob mask". But probably cmake is powerful enough, so this can be expressed in cmake. What do you think?
```
set_property(SOURCE ${CMAKE_CURRENT_LIST_DIR}/pstl_offload/memory/interop_allocs.pass.cpp
    APPEND PROPERTY SRC "pstl_offload/memory/interop_allocs_system.cpp;pstl_offload/memory/interop_allocs_another.cpp")
...
    get_property(additional_src_files SOURCE ${test_source_file} PROPERTY SRC)
    if (DEFINED additional_src_files)
        unset(files)
        foreach(f ${additional_src_files})
            file(GLOB f ${f})
            list(APPEND files ${f})
        endforeach()
        set(additional_src_files ${files})
    endif()
```

Reply:
Yes, this is probably preferable to some extra file to parse.  I was formatting it like that because up until this point, we have relied on the files structure to dictate the tests rather than having some explicit test listing.  My proposed change would keep the CMake code generic and unaware of the specifics of the tests.  However, I have no real preference that it stays that way.

I think since instances of these multi-TU tests are likely rare, it is manageable to list them explicitly in CMake like this.  If you think that many of these tests may exist, then perhaps I would prefer something more generic.

Reply:
I also don't know of an idiom for this.  What you've done here looks reasonable to me.

Reply:
I am not aware of such an idiom as well, but [target_sources](https://cmake.org/cmake/help/latest/command/target_sources.html) may be a good candidate to simplify the logic. It adds source files into the existing target and its dependent targets. 

You can place this after `foreach (_file IN LISTS UNIT_TESTS)` loop. 
```
if (TARGET interop_allocs.pass)
    target_sources(interop_allocs.pass PRIVATE "pstl_offload/memory/interop_allocs_system.cpp")
endif()

```

Still, it is a bit hacky, but it is simpler. 

Probably, it also makes sense to define these customized targets in separate CMakeLists.txt, but I have not investigated this approach yet. I will let you know if I find something.

Reply:
Thanks, this is definitely more clear.